source/common/protobuf/utility.cc

```diff
@@ -645,12 +645,18 @@ ProtobufWkt::Value ValueUtil::listValue(const std::vector<ProtobufWkt::Value>& v
 
 namespace {
 
-void validateDuration(const ProtobufWkt::Duration& duration, int64_t max_seconds_value) {
-  if (duration.seconds() < 0 || duration.nanos() < 0) {
-    throw DurationUtil::OutOfRangeException(
-        fmt::format("Expected positive duration: {}", duration.DebugString()));
+absl::Status validateDurationNoThrow(const ProtobufWkt::Duration& duration,
+                                     int64_t max_seconds_value) {
+  if (duration.seconds() < 0 || duration.nanos() < 0 || duration.nanos() > 999999999 ||
```

Comment:
This should probably be kept split like before (to below 0 and above max-values).

Reply:
Please take al look this comments:  https://github.com/envoyproxy/envoy/pull/27260#discussion_r1189995799

"you do not really need to have different message for negative and too high duration errors"

IIUC, there is no need to have separate check just for throw exception with on says negative duration, the others says too large, since those info is in the durage.DebugString().

Reply:
SG, done!